src/OrbitClientModel/CaptureDeserializer.cpp

```diff
@@ -178,9 +182,36 @@ ErrorMessageOr<CaptureListener::CaptureOutcome> LoadCaptureInfo(
     frame_track_function_ids.insert(function_id);
   }
 
-  capture_listener->OnCaptureStarted(std::move(process), std::move(instrumented_functions),
-                                     std::move(selected_tracepoints),
-                                     std::move(frame_track_function_ids));
+  CaptureStarted capture_started;
+  capture_started.set_process_id(capture_info.process().pid());
+  capture_started.set_executable_path(process.full_path());
+  capture_started.set_process_id(process.pid());
+  capture_started.set_capture_start_timestamp_ns(0);  // Do we have this information anywhere?
+  *capture_started.mutable_modules() = {modules.begin(), modules.end()};
+
+  CaptureOptions* capture_options = capture_started.mutable_capture_options();
+  capture_options->set_pid(capture_info.process().pid());
+  for (const auto& function : capture_info.instrumented_functions()) {
+    InstrumentedFunction instrumented_function;
+    instrumented_function.set_function_id(function.first);
+    instrumented_function.set_function_type(
+        CaptureClient::InstrumentedFunctionTypeFromOrbitType(function.second.orbit_type()));
+
+    instrumented_function.set_file_path(function.second.loaded_module_path());
+    const ModuleData* module_data =
+        module_manager->GetModuleByPath(function.second.loaded_module_path());
```

Comment:
ok, this is the part where we need modules to be available right away. Without it we cannot construct functions. Sending snapshot after capture started event breaks this part.

I think I am going to revert everything now, sorry it did not work out.

Reply:
I don't fully understand so let's discuss offline. Just a couple of notes:
- This feels like it should be solved differently and that we should be able to handle snapshots. One reason is that if we need all modules at the very beginning, this also doesn't work with module updates.
- Maybe this is just a problem of the current capture format that we are moving away from anyway and we can make it work in the new format?